app/Libraries/Notification/BatchIdentities.php

```diff
@@ -46,6 +44,15 @@ public static function fromParams(array $params)
         return $obj;
     }
 
+    public static function scrubIdentities(array $params)
+    {
+        $identities = array_map(function ($param) {
+            return is_array($param) ? static::scrubIdentity($param) : null;
+        }, $params);
+
+        return array_values(array_filter($identities));
```

Comment:
maybe foreach and only add to array if not null instead of this map/filter/values chain?